.github/workflows/backend-workflow.yml

```
@@ -0,0 +1,26 @@
+name: Backend CI
+on: 
+  push:
+    branches-ignore:
+      - 'frontend/**' # Don't run on any frontend branches
+
+jobs:
+  test-backend:
+    name: 'Testing'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: docker-build
+      # This option makes the Postgres container exit when the autoscheduler container does
+        run: docker-compose up --abort-on-container-exit
+
+  lint-backend:
+    name: 'Linting'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: pylint
+        uses: cclauss/GitHub-Action-for-pylint@master
+        with:
+        # Installs Django and pylint-django, then runs pylint on all of the files in autoscheduler/
+          args: pip install -r .github/lint-requirements.txt ; pylint autoscheduler/**/*.py --load-plugins pylint_django

```

It turns out that this doesn't actually get `pylint` to lint *all* of the files, I think it just does files that are one directory down(such as `scraper/banner_requests.py`. For instance, right now it's not linting `autoscheduler/autoscheduler/settings/docker.py`, which has a bunch of linting errors. As such, I think it should be replaced with:

```suggestion
          args: pip install -r .github/lint-requirements.txt ; pylint autoscheduler/** --load-plugins pylint_django
```

We should be able to just try these locally by running the command (just the second half after the semi-colon)

---

That seems to be how it works on my computer as well